internal/plugin.go

```diff
@@ -1621,6 +1613,128 @@ func (p *Plugin) createPlacement(
 	return
 }
 
+// generateClusterSelector determines the type of input and creates a map of selectors to be used in either the
+// clusterSelector or labelSelector field
+func (p *Plugin) generateSelector(
+	resolvedSelectors map[string]interface{},
+	keys []string,
+) (map[string]interface{}, error) {
+	// Determine cluster selector type
+	var selectorType string
+	var usedSelectors []string
+	var legacy bool
+
+	for key := range resolvedSelectors {
+		if key == "matchExpressions" {
+			usedSelectors = append(usedSelectors, key)
+		} else if key == "matchLabels" {
+			usedSelectors = append(usedSelectors, key)
+		} else if !legacy {
+			usedSelectors = append(usedSelectors, "legacy")
+			legacy = true
+		}
+	}
```

Comment:
That doesn't work because the underlying type is always map[string]interface{}:
```go
package main

import "fmt"

func main() {
	temp := map[string]interface{}{"hello": "world"}
	var tempUntyped interface{} = temp
	_, ok := tempUntyped.(map[string]string)
	fmt.Printf("ok: %t\n", ok)
}
```